### PR TITLE
Cleanup Faktory and Category input.

### PR DESCRIPTION
**Pull request by jindrazak**
I cleaned the ArrayCategoryInput. 

Just a question - is there any schema defined for the JSON that gets sent to Algolia? I think it would be nice to have it documented somewhere, so that we know exactly what to send from this app.

### CODE REVIEW

File: src/components/Factory.vue [hunk 1]
```diff
@@ -14,20 +14,26 @@ const newFactory = ref({
     "kategorie2": [],
 })
```
Comment (jindrazak) [inline src/components/Factory.vue:15]: Does the `newFaktory` object has to contain the property `id`? Apparently it's just an index of the factory in the array with no additional value. If it's not required, I'd tend to remove the it from the object.

**Comment by jindrazak** [comment]
Let me fix https://github.com/Smidra/datoun-admin/issues/6 as part of this PR

**Comment by Smidra** [comment]
> I cleaned the ArrayCategoryInput.
> 
> Just a question - is there any schema defined for the JSON that gets sent to Algolia? I think it would be nice to have it documented somewhere, so that we know exactly what to send from this app.

Algolia database is schema-less. https://www.algolia.com/doc/guides/sending-and-managing-data/prepare-your-data/#algolia-records

A sample JSON to send can be found in README.md

File: src/components/Factory.vue (hunk 1 above)
Comment (Smidra) [reply src/components/Factory.vue:15]: It's visible in the fontend. As the label of each factory. Other than that, it's useless.

image

**Comment by Smidra** [comment]
I have reviewed the code and everything looks good to merge.
However, I don't think this closes Issue #6. I cannot remove added Categories out of order just like Aliases.

File: src/components/ArrayCategoryInput.vue [hunk 2]
```diff
@@ -1,96 +1,81 @@
...
+// An array of strings containing categories picked by the user.
+const selectedCategories = ref([""])
 
-function sendArray() {
-    let arrayForSending = []
-    newArray.value.forEach(item => {
-        arrayForSending.push(item.value)
-    });
+// Parsed categories for each category level.
+const categoriesByLevel = computed(() => {
```
Comment (Smidra) [inline src/components/ArrayCategoryInput.vue:12]: Awesome. You have restored the name of my family with this fix.
And I have a lot of Vue to study... :)

**Comment by jindrazak** [comment]
> However, I don't think this closes Issue #6. 

I was't finished with that yet. That's why I marked this PR as "draft". But no worries, I'll create a new PR to fix the Issue.

**Pull request merged**